src/commands/synthetics/utils.ts

```diff
@@ -137,8 +137,12 @@ export const getStrictestExecutionRule = (configRule: ExecutionRule, testRule?:
   return ExecutionRule.BLOCKING
 }
 
-export const hasResultPassed = (result: Result, allowOnUnexpectedResults: boolean): boolean => {
-  if (result.unhealthy && allowOnUnexpectedResults) {
+export const hasResultPassed = (result: Result, failOnCriticalErrors: boolean, failOnTimeout: boolean): boolean => {
+  if (result.unhealthy && failOnCriticalErrors) {
+    return true
+  }
+
+  if (result.error === 'Timeout' && !failOnTimeout) {
```

Comment:
Could you make `result.error` an enum or string union instead so that we do not break it if we change the error message in `createFailingResult`?